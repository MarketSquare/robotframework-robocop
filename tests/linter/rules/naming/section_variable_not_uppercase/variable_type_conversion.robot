*** Variables ***
# Simple type.
${version: float}         7.3
# Parameterized type.
${CRITICAL: list[int]}    [3278, 5368, 5417]
# With @{list} variables the type specified the item type.
@{HIGH: int}              4173    5334    5386    5387
@{HIGH low: int}              4173    5334    5386    5387
# With @{dict} variables the type specified the value type.
&{DATes: date}            rc1=2025-05-08    final=2025-05-15
# Alternative syntax to specify both key and value types.
&{NUMBERS: int=float}     1=2.3    4=5.6
