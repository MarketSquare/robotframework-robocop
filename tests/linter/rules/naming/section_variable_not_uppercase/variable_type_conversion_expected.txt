variable_type_conversion.robot:3:1 [W] NAME08 Section variable '${version: float}' name is not uppercase
variable_type_conversion.robot:8:1 [W] NAME08 Section variable '@{HIGH low: int}' name is not uppercase
variable_type_conversion.robot:10:1 [W] NAME08 Section variable '&{DATes: date}' name is not uppercase